dcor/_dcor.py

```diff
@@ -1169,3 +1174,128 @@ def distance_correlation_af_inv(
             compile_mode=compile_mode,
         ),
     )
+''' 
+A Statistically and Numerically Efficient Independence Test Based on Random Projections and Distance Covariance
+
+url: https://www.frontiersin.org/articles/10.3389/fams.2021.779841/full
+'''
+
+# During implementing u_dustance_covariance_sqr I found this error---------------------------
+# AttributeError: module 'numpy' has no attribute 'bool'.
+# `np.bool` was a deprecated alias for the builtin `bool`. To avoid this error in existing code, use `bool` by itself. Doing this will not modify any behavior and is safe. If you specifically wanted the numpy scalar type, use `np.bool_` here.
+# The aliases was originally deprecated in NumPy 1.20; for more details and guidance see the original release note at:
+#     https://numpy.org/devdocs/release/1.20.0-notes.html#deprecations
+np.bool = np.bool_
+
+
+def gamma_ratio(p): return np.exp(gammaln((p+1)/2) - gammaln(p/2))  # For Calculating C_p and C_q
+
+
+@njit(fastmath=True, parallel=True, cache=True)
+def dist_sum(X):  # It is nothing but the parallelized version of pdist function-----------------------
+    '''
+    Parameters
+    ----------
+    X : 1D array.
+
+    Returns
+    -------
+    res : sum of distinct Euclidean distances corresponding to the elements of X.
+
+    Note: To implement numba, one needs to consider "numpy==1.25" or less. 
+    '''
+    res = 0
+    for i in prange(len(X)):
+        for j in prange(len(X)):
+            if i < j:
+                res += np.abs(X[i] - X[j])
+                pass
+            pass
+        pass
+    return res
+
+
+
+def rndm_projection(X, dim_X):
+    '''
+    Parameters
+    ----------
+    X : N x D, array of arrays
+    where D: number of dimensions (D >= 1) and N: number of samples        
+
+    Returns
+    -------
+    X_new : an array of size N
+    DESCRIPTION: Random projection of multivariate array
+    '''
+
+    # X_std = multivariate_normal.rvs( np.zeros(dim_X), np.identity(dim_X), size = 1)
+    X_std = np.random.standard_normal(dim_X)
+    X_norm = np.sum([i**2 for i in X_std])
+    U_sphere = np.array(X_std / np.sqrt(X_norm))
+    if dim_X > 1:
```

Comment:
The idea of considering == 1 is, one may interest to compute the distance correlation between a univariate and a multivariate data

Reply:
I have taken a look at the thesis of the main author of random projection paper:
https://repository.gatech.edu/server/api/core/bitstreams/27115b8a-5875-445a-a972-b1686799ecb5/content

Based on my understanding from that thesis, the test of independence based on both "dcov(X, Y)(naive)" and "dcov(X, Y)(random projection)" could work on the general data (i.e., dimension of X and Y >=1). The only difference is "dcov(X,Y)(random projection)" based test is faster than "dcov(X,Y)(naive)" test of independence. Now, the question is, can we calculate random projection "dcov (using random projection)" and implement the naive test? Answer is no because one is based on biased dcov estimator and another one is based on unbiased estimator. Moreover, as unbiased dcov does not imply unbiased dcorr (ratio of two unbiased estimators does not guarantee an unbiased estimator of ratio), we cannot even do the multivariate t-test (existing multivariate tests). We have to use a separate test where large sample distribution of the test statistic follows Gamma Distribution (as proposed by the random projection paper).